cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -468,12 +468,14 @@ public void upgradeApplications(FullHttpRequest request, HttpResponder responder
           } catch (UnsupportedOperationException e) {
             String errorMessage = String.format("Application %s does not support upgrade.", appId);
             updateDetail = new ApplicationUpdateDetail(appId, new NotImplementedException(errorMessage));
+            LOG.error(errorMessage, e);
```

Comment:
Understand what situation this exception can get triggered? Generally in http handler, if an error can be created by user input, then we shouldn't log.

Reply:
This is thrown when the app does not support update (Application.isUpdateSupported which is false by default). The user input is namespace and artifactscope. Potentially with a namespace having lot of (user?)apps that does not support upgrade, lot of messages could be generated. I can skip this for now and just keep the logging for generic exception.